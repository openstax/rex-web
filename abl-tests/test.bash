#!/bin/bash

set -e

update_snapshot=$1
my_dir=$(dirname "$0")

if [[ $1 == '--help' ]]; then
    echo "This script runs a test on multiple input files (*.input.json)"
    echo "and compares each output to a corresponding snapshot file (*.snapshot.json)"
    echo ""
    echo "When the output differs from the snapshot the test fails"
    echo ""
    echo "--update-snapshots      Updates all snapshot files"
    exit 1
fi

shopt -s globstar nullglob
for input_file in "$my_dir"/*.input.json; do
    snapshot_file=${input_file%.input.json}.snapshot.json
    output_file=${input_file%.input.json}.temp.json
    if [[ $update_snapshot || ! -f $snapshot_file ]]; then
        output_file=$snapshot_file
    fi
    
    # -------------------------
    # Actual test goes here
    # -------------------------
    json=$(cat "$input_file")
    node "$my_dir/../script/entry.js" transform-approved-books-data --data "$json" > "$output_file"


    diff "$snapshot_file" "$output_file"
    
    if [[ "$snapshot_file" != "$output_file" ]]; then
        rm "$output_file"
    fi
done
shopt -u globstar nullglob