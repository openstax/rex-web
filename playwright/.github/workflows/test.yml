name: CI

on: [push, pull_request]

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: 14
    - run: npm ci
    - run: npx playwright install-deps
    - run: npx playwright install
    - run: npm run ci-coverage
    - run: npm run test

    - name: Coveralls
      uses: coverallsapp/github-action@1.1.3
      with:
        github-token: ${{ secrets.GITHUB_TOKEN }}
