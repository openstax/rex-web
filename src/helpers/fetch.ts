export type FormatError = (status: number, message: string) => Error;

export const rejectResponse = (response: Response, formatError: FormatError) =>
  response.text().then((message: string) => {
    throw formatError(response.status, message);
  });

export const acceptStatus = (
  status: number,
  formatError: FormatError
) => (response: Response): Promise<Response> | Promise<never> => {

  if (response.status !== status) {
    return rejectResponse(response, formatError);
  }
  return Promise.resolve(response);
};
