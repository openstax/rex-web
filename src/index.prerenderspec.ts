/** @jest-environment puppeteer */
import fetch from 'node-fetch';
import { url } from './test/browserutils';

describe('Prerender sanity tests', () => {
  it('has a release manifest', async() => {
    const release = await fetch(url('rex/release.json'))
      .then((response) => response.text());

    expect(release).toMatchSnapshot();
  });
});
