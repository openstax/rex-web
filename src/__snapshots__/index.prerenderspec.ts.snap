// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prerender sanity tests has a release manifest 1`] = `
"{
  \\"archiveUrl\\": \\"/apps/archive/codeversion\\",
  \\"books\\": {
    \\"testbook1-uuid\\": {
      \\"defaultVersion\\": \\"1.0\\"
    }
  },
  \\"code\\": \\"test\\",
  \\"id\\": \\"test\\"
}"
`;